src/dimension.rs

```diff
@@ -66,12 +66,13 @@ pub fn dim_stride_overlap<D: Dimension>(dim: &D, strides: &D) -> bool
 /// to the last element of each dimension should be smaller than the length
 /// of the slice. Also, the strides should not allow a same element to be
 /// referenced by two different index.
-pub fn can_index_slice<A, D: Dimension>(data: &[A],
-                                        dim: &D,
-                                        strides: &D
-                                       ) -> Result<(), StrideError>
+pub fn can_index_slice<A, D: Dimension>(data: &[A], dim: &D, strides: &D)
+    -> Result<(), StrideError>
 {
     if strides.slice().iter().cloned().all(stride_is_positive) {
+        if dim.size_checked().is_none() || strides.size_checked().is_none() {
```

Comment:
I'm wondering if `strides.size_checked().is_none()` is a good check here, since strides have to be interpreted as isize.

Maybe a good check here would be to check if the product of a stride with its corresponding dimension does not overflow.